src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -117,8 +184,11 @@ SGMatrix<float64_t> CGaussianARDKernel::get_parameter_gradient(
 
 float64_t CGaussianARDKernel::distance(int32_t idx_a, int32_t idx_b)
 {
-	REQUIRE(rhs, "Right features (rhs) not set!\n")
 	REQUIRE(lhs, "Left features (lhs) not set!\n")
+	REQUIRE(rhs, "Right features (rhs) not set!\n")
+
+	if (lhs==rhs && idx_a==idx_b) 
```

Comment:
if the left features and right features are the same object, we can use this trick.